int16[]  case
int16[] vel_steps
int16[] qi_move
int16[] q_max
int16 q_depth
---
bool success